File: app/src/main/java/at/shockbytes/dante/network/google/gson/BookBackupSerializer.java
package at.shockbytes.dante.network.google.gson;

import com.google.gson.JsonElement;
import com.google.gson.JsonSerializationContext;
import com.google.gson.JsonSerializer;

import java.lang.reflect.Type;

import at.shockbytes.dante.util.books.Book;


/**
 * @author Martin Macheiner
 *         Date: 13.02.2017.
 */

public class BookBackupSerializer implements JsonSerializer<Book> {


    @Override
    public JsonElement serialize(Book src, Type typeOfSrc, JsonSerializationContext context) {
        return src.toJson();
    }


}

File: app/src/main/java/at/shockbytes/dante/network/google/gson/BookBackupSerializer.kt
package at.shockbytes.dante.network.google.gson

import at.shockbytes.dante.book.BookEntity
import at.shockbytes.dante.util.toJson
import com.google.gson.JsonElement
import com.google.gson.JsonSerializationContext
import com.google.gson.JsonSerializer
import java.lang.reflect.Type


/**
 * @author Martin Macheiner
 * Date: 13.02.2017.
 */

class BookBackupSerializer : JsonSerializer<BookEntity> {

    override fun serialize(src: BookEntity, typeOfSrc: Type,
                           context: JsonSerializationContext): JsonElement = src.toJson()

}
